### ISSUE TITLE
Change Temurin SBOM generation to XML

### ISSUE THREAD

**Issue by andrew-m-leonard** [adoptium/temurin-build#4061]
Currently Temurin produces SBOM's in json format, however this has its limitations and cost:
- json "signing" uses JSF format and very few tooling provides JSF signing and verification. Currently Temurin uses and open source component https://github.com/cyberphone/openkeystore which does not have a huge following, and we have to build and compile this source, which is not provided as an open source binary release.
- Similarly to verify these signatures we need to use the same open source tool, which means "Users" are not easily able to verify and SBOM themselves, without going through a procedure to clone temurin-build and build the TemurinSignSBOM java class that uses openkeystore 
- The [Cyclonedx-cli](https://github.com/CycloneDX/cyclonedx-cli) tool does not provide support for signing and verifying JSON SBOMs:
```
% cyclonedx-cli sign bom test.json --key-file test.pem
Loading private key...
Only XML BOMs are currently supported for signing.
```

As opposed to using an XML format SBOM:
- Provides full feature support for XML, including "signing" and "signature verification"
```
% cyclonedx-cli sign bom testCDXA.xml --key-file cdsa_test.pem 
Loading private key...
Loading XML BOM...
Generating signature...
Saving signature...
% cyclonedx verify all testCDXA.xml --key-file cdsa_test.pem
Loading public key...
Loading XML BOM...
Reading signatures...
Found 1 signatures...
Verifying signature 1... verified

All signatures verified
```
- The [Cyclonedx-cli](https://github.com/CycloneDX/cyclonedx-cli) tool can then easily be used by 3rd party users to both sign and verify SBOMs, CDXAs, ...

**Comment by andrew-m-leonard** [comment]
@netomi hi Thomas, I was wondering what your opinion might be to using XML documents as opposed to JSON, especially when it comes to Eclipse Signing with sigstore potentially?

**Comment by andrew-m-leonard** [comment]
The support for XML SBOM generated using the cyclonedx-core-java is currently blocked due to issue: https://github.com/CycloneDX/cyclonedx-core-java/issues/562

For the moment we will stick with json SBOM until this is fixed, or we have a python-library route...

**Comment by Haroon-Khel** [comment]
@andrew-m-leonard do we want to continue with this now that https://github.com/CycloneDX/cyclonedx-core-java/issues/562 is resolved?

**Comment by andrew-m-leonard** [comment]
> [@andrew-m-leonard](https://github.com/andrew-m-leonard) do we want to continue with this now that [CycloneDX/cyclonedx-core-java#562](https://github.com/CycloneDX/cyclonedx-core-java/issues/562) is resolved?

Yes, all the necessary CycloneDX issues are fixed. So we can move forward with this, I will add PMCAgenda to propose

**Comment by andrew-m-leonard** [comment]
I will get user feedback on JSON vs XML...

**Comment by andrew-m-leonard** [comment]
@Haroon-Khel following PMC review, we want to get user input and thoughts on changing to xml SBOMs first, before we change
